lib/constable_web/models/announcement.ex

```diff
@@ -16,7 +16,7 @@ defmodule Constable.Announcement do
     belongs_to :user, User
     has_many :comments, Comment, on_delete: :delete_all
     has_many :subscriptions, Subscription, on_delete: :delete_all
-    many_to_many :interests, Interest, join_through: AnnouncementInterest
+    many_to_many :interests, Interest, on_delete: :delete_all, join_through: AnnouncementInterest
```

Comment:
Do you know if this deletes all the associated `announcements_interests` records associated or does it delete all of the associated `interests` as well? We probably want the former but definitely not the latter, right?

Reply:
It seems I should've done some reading before. It seems like it won't delete interests, 

> If you attempt to remove associated many_to_many data, Ecto will always remove data from the join schema and never from the target associations be it by setting :on_replace to :delete, :on_delete to :delete_all or by using changeset functions such as Ecto.Changeset.put_assoc/3. For example, if a Post has a many to many relationship with Tag, setting :on_delete to :delete_all will only delete entries from the “posts_tags” table in case Post is deleted.

from https://hexdocs.pm/ecto/Ecto.Schema.html#many_to_many/3-removing-data

Reply:
Correct, it doesn't delete the interests, just the association.